webpack/components/NewTemplateSync/components/NewTemplateSyncForm/NewTemplateSyncForm.js

```diff
@@ -1,158 +1,105 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import { compose } from 'redux';
 
 import ForemanForm from 'foremanReact/components/common/forms/ForemanForm';
-import * as Yup from 'yup';
+import { useForemanLocation, useForemanOrganization } from 'foremanReact/Root/Context/ForemanContext';
+
 import SyncSettingsFields from '../SyncSettingFields';
 import SyncTypeRadios from '../SyncTypeRadios';
-
-const redirectToResult = history => () =>
-  history.push({ pathname: '/template_syncs/result' });
-
-const repoFormat = formatAry => value => {
-  if (value === undefined) {
-    return true;
-  }
-
-  const valid = formatAry
-    .map(item => value.startsWith(item))
-    .reduce((memo, item) => item || memo, false);
-
-  return value && valid;
-};
-
-const syncFormSchema = (syncType, settingsObj, validationData) => {
-  const schema = (settingsObj[syncType].asMutable() || []).reduce(
-    (memo, setting) => {
-      if (setting.name === 'repo') {
-        return {
-          ...memo,
-          repo: Yup.string()
-            .test(
-              'repo-format',
-              `Invalid repo format, must start with one of: ${validationData.repo.join(
-                ', '
-              )}`,
-              repoFormat(validationData.repo)
-            )
-            .required("can't be blank"),
-        };
-      }
-      return memo;
-    },
-    {}
-  );
-
-  return Yup.object().shape({
-    [syncType]: Yup.object().shape(schema),
-  });
-};
-
-class NewTemplateSyncForm extends React.Component {
-  allowedSyncType = (userPermissions, radioAttrs) =>
-    this.props.userPermissions[radioAttrs.permission];
-
-  constructor(props) {
-    super(props);
-
-    this.radioButtons = [
-      { label: 'Import', value: 'import', permission: 'import' },
-      { label: 'Export', value: 'export', permission: 'export' },
-    ];
-
-    this.state = {
-      syncType: this.radioButtons.find(radioAttrs =>
-        this.allowedSyncType(props.userPermissions, radioAttrs)
-      ).value,
-    };
-  }
-
-  updateSyncType = event => {
-    this.setState({ syncType: event.target.value });
+import { redirectToResult, repoFormat, syncFormSchema } from './NewTemplateSyncFormHelpers';
+
+const NewTemplateSyncForm = ({
+  error,
+  submitForm,
+  importSettings,
+  exportSettings,
+  history,
+  validationData,
+  importUrl,
+  exportUrl,
+  initialValues,
+  userPermissions
+}) => {
+  const allowedSyncType = (currentUserPermissions, radioAttrs) =>
+    currentUserPermissions[radioAttrs.permission];
+
+  const radioButtons = [
+    { label: 'Import', value: 'import', permission: 'import' },
+    { label: 'Export', value: 'export', permission: 'export' },
```

Comment:
Not related to this pr but just something i noticed, i think the labels should be marked for translation:
```suggestion
    { label: __('Import'), value: 'import', permission: 'import' },
    { label: __('Export'), value: 'export', permission: 'export' },
```